### PR TITLE
Move parseMany benchamrks from FileSystem.Handle to Data.Parser

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: benchmark/Streamly/Benchmark/Data/Parser.hs [hunk 1]
```diff
@@ -18,21 +19,30 @@ import Data.Foldable (asum)
 import Data.Functor (($>))
 import Data.Maybe (fromMaybe)
 import Data.Monoid (Sum(..))
+import GHC.Magic (inline)
+#if __GLASGOW_HASKELL__ >= 802
+import GHC.Magic (noinline)
+#else
+#define noinline
```
Comment (adithyaov) [inline benchmark/Streamly/Benchmark/Data/Parser.hs:26]: This was there in `FileSystem.Read`, what does this do?

Comment (adithyaov) [inline benchmark/Streamly/Benchmark/Data/Parser.hs:26]: What is the purpose of this?

Comment (harendra-kumar) [reply benchmark/Streamly/Benchmark/Data/Parser.hs:26]: it defines "noinline" to empty value. Wherever "noinline" occurs in the file it will be erased.

**Review by harendra-kumar — APPROVED** [review]

File: benchmark/Streamly/Benchmark/Data/Parser.hs (hunk 1 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/Data/Parser.hs:26]: Ah. Very sneaky.

**Pull request merged**